pkg/internal/client/txn.go

```diff
@@ -706,6 +707,11 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	// If we're not yet writing in this txn, but intend to, insert a
 	// begin transaction request before the first write command.
 	if needBeginTxn {
+		// If we're setting the system-config trigger, anchor the transaction to
```

Comment:
That is very fragile. Please take a look at what you're suggesting (i.e. try and figure out what we would need to modify in the `sql` package). I did consider adding a `Txn.SetTxnKey()` method so that the user could specify how they wanted a transaction anchor, but that felt overly specific too. If we require the system-config trigger to be associated with a transaction anchored on the system-config range, why not make that automatic?